### ISSUE TITLE
Fail to create TLSGateway from DomainMapping's Kingress due to invalid name

### ISSUE THREAD

**Issue by nak3** [knative-sandbox/net-istio#514]
### Description

`serving-https test` is always failing on `TestDomainMapping`. Please refer to any [log](https://prow.knative.dev/view/gcs/knative-prow/logs/ci-knative-serving-https/1359437522937057280) in `https` test in [test grid](https://testgrid.k8s.io/knative-serving#knative-serving-https).

This is because net-istio tries to create gateway name with "dot" like `domain-mapping-hfyjwokc.example.org-3797421420`.

```
serving-tests                          39m         Warning   InternalError                  ingress/domain-mapping-hfyjwokc.example.org                                             admission webhook "validation.istio.io" denied the request: configuration is invalid: invalid gateway name: "domain-mapping-hfyjwokc.example.org-3797421420"
```

#### For more detail...

- Gateway name is created from KIngress's name:

https://github.com/knative-sandbox/net-istio/blob/3bba898d0fed6640de929e3ff8f8bcaddb66bdee/pkg/reconciler/ingress/resources/gateway.go#L265-L268

- Istio does not allow dots.

https://github.com/istio/istio/blob/b9545357f0d57ed31d24a8ad8ea7a3cea883f63a/pkg/config/validation/validation.go#L334-L337

### Steps to produce

- Please refer to https://testgrid.k8s.io/knative-serving#knative-serving-https
- Or you can re-create the issue with autoTLS & DomainMapping.

**Comment by ZhiminXiang** [comment]
So the Gateway name has dot because the name of the Kingress has dot. 
Maybe we should check if `accessor.GetName()` is valid. If it is invalid, we replace it with a valid format.

**Comment by tcnghia** [comment]
Fixed by #532
